kafka/producer/producer_batch.go

```diff
@@ -96,3 +103,16 @@ func (b *producerBatch) FlushMessages() {
 	}
 	b.dcpCheckpointCommit()
 }
+
+func isPermanentError(err error) bool {
```

Comment:
we could consider renaming this method as "Fatal" because it deals with errors that cannot be fixed. We can easily understand that a "Permanent Error" should be considered "Fatal," even though we don't directly say it.